### PR TITLE
Adds Support for Push Notifications

### PR DESCRIPTION
**Pull request by Phillip3**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -52,10 +53,62 @@ if (
...
+  OneSignal.inFocusDisplaying(2); // Controls what should happen if a notification is received while the app is open. 2 means that the notification will go directly to the device's notification center.
+
+  // The promptForPushNotifications function code will show the iOS push notification prompt. We recommend removing the following code and instead using an In-App Message to prompt for notification permission (See step below)
+  OneSignal.promptForPushNotificationsWithUserResponse(myiOSPromptCallback);
+
+  // componentWillUnmount() {
+  //   OneSignal.removeEventListener('received', this.onReceived);
+  //   OneSignal.removeEventListener('opened', this.onOpened);
+  //   OneSignal.removeEventListener('ids', this.onIds);
+  // }
```
Comment (dereckquock) [inline App.js]:
```suggestion
```

File: App.js [hunk 2]
```diff
@@ -52,10 +53,62 @@ if (
...
+  useEffect(() => {
```
Comment (dereckquock) [inline App.js]: are we doin anything with these event handlers?

File: android/app/build.gradle [hunk 3]
```diff
@@ -83,7 +94,7 @@ project.ext.react = [
 
 apply from: '../../node_modules/react-native-unimodules/gradle.groovy'
 apply from: "../../node_modules/react-native/react.gradle"
-apply from: "../../node_modules/expo-updates/expo-updates.gradle"
```
Comment (dereckquock) [inline android/app/build.gradle:86]: is this supposed to be removed?

File: App.js [hunk 4]
```diff
@@ -52,10 +53,62 @@ if (
...
+  // Replace 'YOUR_ONESIGNAL_APP_ID' with your OneSignal App ID.
+  OneSignal.init(keys.ONESIGNAL_APP_ID, {
+    kOSSettingsKeyAutoPrompt: false,
+    kOSSettingsKeyInAppLaunchURL: false,
+    kOSSettingsKeyInFocusDisplayOption: 2,
+  });
+  OneSignal.inFocusDisplaying(2); // Controls what should happen if a notification is received while the app is open. 2 means that the notification will go directly to the device's notification center.
+
+  // The promptForPushNotifications function code will show the iOS push notification prompt. We recommend removing the following code and instead using an In-App Message to prompt for notification permission (See step below)
+  OneSignal.promptForPushNotificationsWithUserResponse(myiOSPromptCallback);
```
Comment (dereckquock) [inline App.js]: are we doin anything with the callback?

**Review by dereckquock — CHANGES_REQUESTED** [review]
awesomeeeee 🔥

File: App.js (hunk 4 above)
Comment (Phillip3) [reply App.js]: We are not! At let I'm not sure we need to be. Do we need to amplitude log that?

File: App.js (hunk 2 above)
Comment (Phillip3) [reply App.js]: I don't think I understand your question, but here's some info:

We need the event listeners so we can receive push notifications while the app is active, but they should be removed before unloading the application so we're not leaking memory.

File: android/app/build.gradle (hunk 3 above)
Comment (Phillip3) [reply android/app/build.gradle:86]: YOURE SUPPOSED TO BE REMOVED.

Comment (Phillip3) [reply android/app/build.gradle:86]: Also, yes. This was intentional. I changed the path because of an update in the package files 😄

**Review by dereckquock — APPROVED** [review]
🔥

**Pull request merged**